src/nvim/charset.c

```diff
@@ -1240,7 +1250,7 @@ void getvcol(win_T *wp, pos_T *pos, colnr_T *start, colnr_T *cursor,
           if (enc_utf8 && (c >= 0x80)) {
             incr = utf_ptr2cells(ptr);
           } else {
-            incr = CHARSIZE(c);
+            incr = char_size(c);
```

Comment:
Ah, CMakeLists.txt add -Wconversion unconditionally and then it is filtered out in src/nvim/CmakeLists.txt. I have only checked top-level CMakeLists.txt. (By the way, how about disabling GCC -Wconversion? It constantly produces warnings cmake does not in cases when I do not see any problems.)